gemd/units/impl.py

```diff
@@ -26,6 +26,9 @@ def _scaling_preprocessor(input_string: str) -> str:
     tight_division = False
     scales = []
 
+    if next(token for token in tokens).type == NUMBER:
+        return input_string  # The unit can't have a leading number; scaling factors are internal
```

Comment:
Why does returning the input string lead to the desired error being thrown?

Also, why is a leading number not allowed? If `g / 100 L` is then why not `100 L / g`? Or is this to cover for the cases in which the input string is _just_ a number?

Reply:
We exit the preprocessor without mangling the leading digit, so it's handled as it would have been handled by default.  We could probably get equivalent functionality by just swallowing the first term, but I didn't want to have to deal with possible weirdness following from that if that first term would have set some of the internal booleans.

No leading digit because normal Pint behavior says that should be a magnitude on a value, and that kinda matches.  I personally also can't think of a customary unit that has a scaling factor that has it up front like that.  It conveniently also covers the case where it's just a number.